ansible/roles-infra/infra-osp-project-create/tasks/main.yaml

```diff
@@ -1,79 +1,118 @@
 ---
-- name: Create project and quota
-  when: osp_project_create
+- name: Generate user password
+  set_fact:
+    heat_user_password_generated: "{{lookup('password', '/dev/null length=12 chars=ascii_letters,digits')}}"
+
+- name: Set heat_user_password
+  set_fact:
+    heat_user_password: "{{ heat_user_password_provided | default(heat_user_password_generated) }}"
+
+- name: Set fact for new user password
+  set_fact:
+    osp_auth_password_member: "{{ heat_user_password_provided | default(heat_user_password_generated) }}" 
+
+- name: Set fact for new user
+  set_fact:
+    osp_auth_username_member: "{{ guid }}-user"
+
+- when: >-
+    osp_project_create | bool
+    or osp_project_id is not defined
+    or osp_create_sandbox | bool
+  environment:
+    OS_AUTH_URL: "{{ osp_auth_url }}"
+    OS_USERNAME: "{{ osp_auth_username }}"
+    OS_PASSWORD: "{{ osp_auth_password }}"
+    OS_PROJECT_NAME: "admin"
+    OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
+    OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
   block:
-    - name: Create project for user
-      os_project:
-        name: "{{ osp_project_name }}"
-        state: present
-        description: "{{ env_type }} for user {{ student_name }}"
-        enabled: True
-        domain_id: "Default"
-        endpoint_type: public
+  - name: Create project and quota
+    when: osp_project_create
+    block:
+      - name: Create project for user
+        os_project:
+          name: "{{ osp_project_name }}"
+          state: present
+          description: "{{ env_type }} for user {{ student_name }}"
+          enabled: True
+          domain_id: "Default"
+          endpoint_type: public
 
-    - name: Grant access to admin account on new project
-      os_user_role:
-        state: present
-        user: "{{ admin_user }}"
-        role: "admin"
-        project: "{{ osp_project_name }}"
+      - name: Grant access to admin account on new project
+        os_user_role:
+          state: present
+          user: "{{ admin_user }}"
+          role: "admin"
+          project: "{{ osp_project_name }}"
+
+      - name: Set quotas on new project
+        os_quota:
+          name: "{{ osp_project_name }}"
+          instances: "{{ quota_num_instances }}"
+          cores: "{{ quota_num_cores }}"
+          ram: "{{ quota_memory }}" #in MB
+          volumes: "{{ quota_num_volumes }}"
+          gigabytes: "{{ quota_volumes_gigs }}" #volume storage
+          #loadbalancer: #when Octavia is available
+          #pool: #when Octavia is available
+          network: "{{ quota_networks }}"
+          subnet: "{{ quota_subnets }}"
+          router: "{{ quota_routers }}"
+          floatingip: "{{ quota_fip }}"
+          security_group: "{{ quota_sg }}"
+          security_group_rule: "{{ quota_sg_rules }}"
+
+      - name: Create user in new project
+        os_user:
+          state: present
+          name: "{{ guid }}-user"
```

Comment:
shouldn't it be `osp_auth_username_member` ?

Reply:
yar, updated